File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n = cardPoints.length;
        int[] prefixSum = new int[k + 1];
        int[] suffixSum = new int[k + 1];
        int maxScore = 0;

        // compute prefixSum
        for(int i = 0; i < k ;i++){
            prefixSum[i + 1] = prefixSum[i] + cardPoints[i];
        }

        // compute suffixSum
        for( int j = 0; j < k;j++){
            suffixSum[j + 1] = suffixSum[j] + cardPoints[n - j - 1];
        }

        // find maxScore
        for(int x = 0; x <= k; x++){
            maxScore = Math.max( prefixSum[x] + suffixSum[k - x], maxScore);
        }
        return maxScore;

    }
}

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n = len(cardPoints)
        total_sum = sum(cardPoints[:k])
        maxSum = total_sum
        for i in range(1,k+1):
            total_sum = total_sum - cardPoints[k - i] + cardPoints[n - i]
            maxSum = max(maxSum,total_sum)
        return maxSum
        